test/unit/kernel/src/KernelWithAdditionalParam.cpp

```diff
@@ -48,13 +47,98 @@ void operator()()
     alpaka::test::KernelExecutionFixture<TAcc> fixture(
         alpaka::vec::Vec<Dim, Idx>::ones());
 
-    KernelWithAdditionalParam kernel;
+    KernelWithAdditionalParamByValue kernel;
 
     REQUIRE(fixture(kernel, 42));
-}
+  }
 };
 
-TEST_CASE( "KernelWithAdditionalParam", "[kernel]")
+TEST_CASE("KernelWithAdditionalParamByValue", "[kernel]")
 {
-    alpaka::meta::forEachType< alpaka::test::acc::TestAccs >( TestTemplate() );
+    alpaka::meta::forEachType<alpaka::test::acc::TestAccs>(TestTemplateByValue());
 }
+
+/*
+Passing a parameter by non-const reference is not allowed.
```

Comment:
Maybe change `non-const reference` to the standard formulation `reference to non-const`?

Reply:
done